New Verify API | Changed "account/list"

ArkShocer opened the pull request:
These changes add the functionality to create a web-app for users / administrators to login and manage the servers.
	
Tasks:
- [x] Add API to verify password
- [x] Change output of `account/list` call `usernames` instead of `ids`
- [ ] Add API Key in `Header` to instances
	- [ ] Instances can't connect to master server due to missing api key in header
- [x] Deny access to api calls without key
- [x] Exception for /verify
	- [x] Respond with 401 if no key
	- [x] Verify Response
		- [ ] Respond with key from `config.xml`
- [x] Add New XmlElement `Key` to UchuConfiguration
	- [x] Add `Key` from `config.xml` to `ApiManager.cs`
	- [ ] Add `Key` from `config.xml` to `AccountCommands.cs`
- [x] Set `Access-Control-Allow-Origin, *` for all responses
- [ ] Fix https protocol not working
	- [ ] All instances stop working if changing `http` to `https` in `config.xml`

ArkShocer committed:
New Verify API | Changed "account/list"

- Added Password Verify API - Current NodeJS packages don't support the "EnhancedHashPassword" function, because of that, I would need an API to verify with the hashed passwords.
- Changed "account/list" to output Usernames instead of Ids

ArkShocer committed:
Merge branch 'UchuServer:dev' into dev

ArkShocer committed:
Update API.md

- Added Account Verify API

ArkShocer committed:
Update API.md

TheNexusAvenger commented:
What is the justification/use case for these changes?

TecCheck commented:
I think the verify API could be handy for logging into a web panel like the one Darkflame has.

ArkShocer commented:
As @TecCheck  mentioned, this API change adds the ability to verify user credentials over http which can then be used to create a web based interface for users/administrators to login to and see their stats/manage the server.

ArkShocer committed:
Implemented requirement of Key

- Added Key to XmlElement, Verify Response
- Set ContentType, ContentEncoding, Acccess-Control-Allow-Origin to all responses
- Added 401 Response if no header / wrong header token is present

ArkShocer committed:
Added `Key` to `ApiManager.cs`

Thanks to zaop for the help
